### ISSUE TITLE
import type can't be used with interface extends

### ISSUE THREAD

**Issue by mjbvz** [microsoft/TypeScript#36478]
**TypeScript Version:**  3.8.0-dev.20200128

**Search Terms:**
- import type
- extends interface
- 

**Code**
For the TS:

```ts
import type * as iface from './iface';

interface Foo extends iface.IFace { }
```

**Expected behavior:**
No errors. `iface` is only being used as a type, not a value

**Actual behavior:**
Error:

```
'iface' cannot be used as a value because it was imported using 'import type'
```

This code previously did not produce errors in TS 3.8-beta

**Related issues**
- #36428

/cc @andrewbranch

**Comment by bolinfest** [comment]
This is marked "closed," but I am still seeing this issue on TypeScript 3.8.3.

**TypeScript Version:**  3.8.3


**Search Terms:** import type

**Code**

```ts
import type * as monaco from "monaco-editor";

// Trying to create a subtype of IViewZone that provides
// a guarantee that marginDomNode is not null.
interface MyViewZone extends monaco.editor.IViewZone {
  marginDomNode: HTMLElement;
}
```

**Expected behavior:**
This should typecheck.

**Actual behavior:**
```
error TS1361: 'monaco' cannot be used as a value because it was imported using 'import type'.

24 interface MyViewZone extends monaco.editor.IViewZone {
                                      ~~~~~~

  src/zoneManager.ts:5:13
    5 import type * as monaco from "monaco-editor";
                  ~~~~~~~~~~~
    'monaco' was imported here.
```

Note that in the same file, I can do:

```
type ZoneWidget = {
  afterLineNumber: number;
  zone: monaco.editor.IViewZone;
};
```

so it is surprising that I can use `monaco.editor.IViewZone` in one place but not the other.


**Playground Link:**
Unfortunately, I cannot repro in a playground as I would expect. Perhaps this is something specific to `monaco-editor`? For example, doing something analogous with the `http` Node built-in works fine:

https://www.typescriptlang.org/play/?ssl=1&ssc=1&pln=5&pc=2#code/JYWwDg9gTgLgBDAnmApnAVHAhgZzgCxhjDgDMoIQ4AiQ46gbgCgngA7GFKUrAYzQCyiAMIAbYCg4AlFAEcArihwwAglADmeFAA9ObACZ46YAHRiJ0uYuVrNcAN5M4zuGAowIvCKIBccZVDs6swAvkA

**Related Issues:** This one!

**Comment by bolinfest** [comment]
I suspect it has to do with the depth into the namespace? It appears I could do this:

```ts
import type * as monaco from "monaco-editor";

interface MyViewZone extends monaco.IRange {
  marginDomNode: HTMLElement;
}
```

but not this:

```ts
import type * as monaco from "monaco-editor";

interface MyViewZone extends monaco.editor.IViewZone {
  marginDomNode: HTMLElement;
}
```

**Comment by bolinfest** [comment]
Ah, though as a workaround I can do:

```ts
import type * as monaco from "monaco-editor";
type IViewZone = monaco.editor.IViewZone;

interface MyViewZone extends IViewZone {
  marginDomNode: HTMLElement;
}
```
